### ISSUE TITLE
False negative on class inheritance with tuple unpack

### ISSUE THREAD

**Issue by luwqz1** [microsoft/pyright#9259]
```python
class Foo(*(int, object)): ... # no error

Foo(1).real  # unknown
```

Unlike that, mypy is working correctly in [mypy playground](https://mypy-play.net/?mypy=latest&python=3.13&gist=a8ab44fd7263418c182047c5f5522e46)

**Comment by erictraut** [comment]
Pyright is working as intended here. The code above runs without error at runtime, so pyright does not (nor should it) generate a static type checking error in standard type checking mode. If you enable strict mode, you'll see that it generates two errors indicating that the "base class type is unknown, obscuring type of derived class" and the "type of 'real' is Unknown". If you'd like to see these stricter type checking errors, feel free to enable these diagnostic checks or enable strict mode.

**Comment by luwqz1** [comment]
> Pyright is working as intended here. The code above runs without error at runtime, so pyright does not (nor should it) generate a static type checking error in standard type checking mode. If you enable strict mode, you'll see that it generates two errors indicating that the "base class type is unknown, obscuring type of derived class" and the "type of 'real' is Unknown". If you'd like to see these stricter type checking errors, feel free to enable these diagnostic checks or enable strict mode.

i got it, thanks!
